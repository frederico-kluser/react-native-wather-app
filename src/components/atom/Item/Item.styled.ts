import styled from 'styled-components/native';
import {itemSize} from '../../../helpers/dimentions';

export const Container = styled.TouchableOpacity`
  border-radius: 16px;
  background-color: #fff;
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  box-sizing: border-box;
  height: ${itemSize}px;
  ${({marginRight}) => marginRight && 'margin-right: 32px;'}
  margin-bottom: 32px;
  padding: 16px;
  width: ${itemSize}px;
`;

export const Temperature = styled.Text`
  font-size: 37px;
`;

export const City = styled.Text`
  font-size: 12px;
  font-weight: 500;
  margin-top: 8px;
`;

export const Country = styled.Text`
  font-size: 12px;
  margin-top: 6px;
`;

export const WeatherIcon = styled.Image`
  height: 64px;
  position: absolute;
  right: 10px;
  top: 10px;
  width: 64px;
`;

export const WeatherInfo = styled.View`
  flex-direction: row;
  position: absolute;
  ${({rightPosition}) => (rightPosition ? 'right' : 'left')}: 16px;
  bottom: 16px;
`;

export const WeatherInfoIcon = styled.Image`
  width: 15px;
  height: 15px;
`;

export const WeatherInfoText = styled.Text`
  font-size: 12px;
  font-weight: bold;
`;

export default Container;
