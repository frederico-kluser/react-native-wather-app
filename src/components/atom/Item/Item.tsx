import React from 'react';
import {IconProps, weatherIcons} from '../../../styles/icons';
import Container, {
  City,
  Country,
  Temperature,
  WeatherIcon,
  WeatherInfo,
  WeatherInfoIcon,
  WeatherInfoText,
} from './Item.styled';

export interface ItemInterface {
  temperature: number;
  func: Function;
  index: number;
  city: string;
  country: string;
  status: IconProps;
}

const Item = ({
  city,
  country,
  func = () => {},
  humidity,
  index,
  status,
  temperature,
  wind,
}: ItemInterface) => (
  <Container marginRight={index % 2 === 0} onPress={() => func(index)}>
    <WeatherIcon source={weatherIcons[status]} />
    <Temperature>{temperature}Â°</Temperature>
    <City>{city}</City>
    <Country>{country}</Country>
    <WeatherInfo>
      <WeatherInfoIcon source={weatherIcons.humidity} />
      <WeatherInfoText>{humidity}%</WeatherInfoText>
    </WeatherInfo>
    <WeatherInfo rightPosition>
      <WeatherInfoIcon source={weatherIcons.wind} />
      <WeatherInfoText>{wind} km/h</WeatherInfoText>
    </WeatherInfo>
  </Container>
);

export default Item;
