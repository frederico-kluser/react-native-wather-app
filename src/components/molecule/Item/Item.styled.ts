import styled from 'styled-components/native';

const Container = styled.TouchableOpacity`
  background-color: #fff;
  border: 2px solid #000;
  border-radius: 16px;
  height: 100px;
  padding: 16px;
  position: relative;
  width: 100%;
  margin-bottom: ${({lastElement}) => (lastElement ? '0px' : '16px')};
`;

export default Container;
