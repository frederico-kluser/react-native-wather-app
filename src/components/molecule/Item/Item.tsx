import React from 'react';
import Title from '../../atom/Title/Title';
import Container from './Item.styled';

const Item = ({index, name, func = () => {}, lastElement}: any) => (
  <Container lastElement={lastElement} onPress={() => func(index)}>
    <Title text={name} />
  </Container>
);

export default Item;
