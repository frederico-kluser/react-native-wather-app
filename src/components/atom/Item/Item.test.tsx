import React from 'react';
import renderer from 'react-test-renderer';
import Item, {ItemInterface} from './Item';

const mockedProps: ItemInterface = {
  city: 'SÃ£o Paulo',
  func: () => {},
  index: 0,
  country: 'Brazil',
  temperature: 10,
  status: 'day',
};

test('renders correctly', () => {
  const tree = renderer.create(<Item {...mockedProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
