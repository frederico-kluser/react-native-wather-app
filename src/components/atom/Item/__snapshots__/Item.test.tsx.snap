// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 16,
      "boxSizing": "border-box",
      "height": 343,
      "marginBottom": 32,
      "marginRight": 32,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "shadowColor": "rgba(0, 0, 0, 0.24)",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 8,
      "width": 343,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "height": 64,
          "position": "absolute",
          "right": 10,
          "top": 10,
          "width": 64,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 37,
        },
      ]
    }
  >
    10
    °
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "500",
          "marginTop": 8,
        },
      ]
    }
  >
    São Paulo
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 12,
          "marginTop": 6,
        },
      ]
    }
  >
    Brazil
  </Text>
  <View
    style={
      Array [
        Object {
          "bottom": 16,
          "flexDirection": "row",
          "left": 16,
          "position": "absolute",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/humidity.png",
        }
      }
      style={
        Array [
          Object {
            "height": 15,
            "width": 15,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
          },
        ]
      }
    >
      %
    </Text>
  </View>
  <View
    rightPosition={true}
    style={
      Array [
        Object {
          "bottom": 16,
          "flexDirection": "row",
          "position": "absolute",
          "right": 16,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/wind.png",
        }
      }
      style={
        Array [
          Object {
            "height": 15,
            "width": 15,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
          },
        ]
      }
    >
       km/h
    </Text>
  </View>
</View>
`;
